lib/Subscriber/Phpcr/RemoveSubscriber.php

```diff
@@ -55,12 +56,20 @@ public static function getSubscribedEvents()
      * remove any references to the node.
      *
      * @param RemoveEvent $event
+     *
+     * @throws DocumentReferencedException
      */
     public function handleRemove(RemoveEvent $event)
     {
         $document = $event->getDocument();
         $node = $this->documentRegistry->getNodeForDocument($document);
 
+        $references = iterator_to_array($node->getReferences());
```

Comment:
Well, I guess we could move that check to sulu/sulu, but IMO that's an issue also the document manager should be caring about.

However, because of some bugs in jackalope I can't continue to fix this the proper way, so this PR will stay open. Have fixed it in a less cleaner way now, and created an issue for later processing: https://github.com/sulu-io/sulu/issues/1710